tests/Equinox.EventStore.Integration/StoreIntegration.fs

```diff
@@ -365,7 +367,53 @@ type Tests(testOutputHelper) =
         let! _ = service.Read cartId
         test <@ singleBatchBackwards @ batchBackwardsAndAppend @ singleBatchBackwards = capture.ExternalCalls @>
     }
-#endif
+    #else
+    [<AutoData(SkipIfRequestedViaEnvironmentVariable="EQUINOX_INTEGRATION_SKIP_EVENTSTORE")>]
+    let ``Can roundtrip against Store, correctly snapshotting to avoid redundant reads`` (ctx, skuId) = Async.RunSynchronously <| async {
+        #if NET
+        use _ = source.StartActivity("Can roundtrip against Store, correctly snapshotting to avoid redundant reads")
+        #endif
+        let log, capture = output.CreateLoggerWithCapture()
+        let! client = connectToLocalStore log
+        let batchSize = 10
+        let context = createContext client batchSize
+        let service = Cart.createServiceWithCompaction log context
+
+        // Trigger 8 events, then reload
+        let cartId = % Guid.NewGuid()
+        do! addAndThenRemoveItemsManyTimes ctx cartId skuId service 4
+        let! _ = service.Read cartId
+
+        test <@ readSnapshotted @ [EsAct.Append] @ readSnapshotted = capture.ExternalCalls @>
+
+        // Add two more, which should push it over the threshold and hence trigger an append of a snapshot event
+        capture.Clear()
+        do! addAndThenRemoveItemsManyTimes ctx cartId skuId service 1
+        test <@ readSnapshotted @ [EsAct.Append; EsAct.Append] = capture.ExternalCalls @>
+
+        // We now have 10 events and should be able to read them with a single call
+        capture.Clear()
+        let! _ = service.Read cartId
+        test <@ readSnapshotted = capture.ExternalCalls @>
+
+        // Add 8 more; total of 18 should not trigger snapshotting as we snapshotted at Event Number 10
+        capture.Clear()
+        do! addAndThenRemoveItemsManyTimes ctx cartId skuId service 4
+        test <@ readSnapshotted @ [EsAct.Append] = capture.ExternalCalls @>
+
+        // While we now have 18 events, we should be able to read them with a single call
+        capture.Clear()
+        let! _ = service.Read cartId
+        test <@ readSnapshotted = capture.ExternalCalls @>
+
+        // add two more events, triggering a snapshot, then read it in a single snapshotted read
+        capture.Clear()
+        do! addAndThenRemoveItemsManyTimes ctx cartId skuId service 1
+        // and reload the 20 events with a single read
+        let! _ = service.Read cartId
+        test <@ readSnapshotted @ [EsAct.Append; EsAct.Append] @ readSnapshotted = capture.ExternalCalls @>
```

Comment:
should there be assertions of how many events there are in the snapshots stream ?
I think not, but not 100% sure

Reply:
I'd be happy to leave that assertion out as ultimately the number of events doesn't matter, we only read the last one, as long as it is appended in one go (which is asserted on)

Reply:
sounds about right, I guess the best thing to predicate things on is whether the correct number of slice reads happen (exactly one) based on the intended timing of when snapshots get written.